src/CodeGen_C.cpp

```diff
@@ -2151,6 +2163,22 @@ void CodeGen_C::visit(const Call *op) {
         user_error << "Signed integer overflow occurred during constant-folding. Signed"
             " integer overflow for int32 and int64 is undefined behavior in"
             " Halide.\n";
+    } else if (op->is_intrinsic(Call::quiet_div)) {
+        internal_assert(op->args.size() == 2);
+        if (is_zero(op->args[1])) {
+            user_error << "Division by constant zero in expression: " << Expr(op) << "\n";
```

Comment:
This behavior is different than CodeGen_LLVM... intended?

https://github.com/halide/Halide/blob/760832046efee7587822052fcf0e1b7c3abb02e9/src/CodeGen_LLVM.cpp#L2756

Reply:
It behaves that way because it is assuming the codegen around it has assured it can never happen.

Reply:
Ah, so in that case it should probably be `internal_error`

Reply:
And in CodeGen_LLVM too then?

Reply:
Done

Reply:
The technique being used in LLVM codegen is what I would expect for some case where the code at runtime can never hit the case where the division by zero executes. Translating that into an internal error requires that the compiler is good enough to remove the IR containing division.

Reply:
...in which case I should back out the last two changes (and have the C backend just use '0' or other arbitrary for the undef case, I suppose).